validation/src/main/scala/hmda/validation/engine/lar/syntactical/LarSyntacticalEngine.scala

```diff
@@ -6,19 +6,19 @@ import hmda.validation.context.ValidationContext
 import hmda.validation.engine.Syntactical
 import hmda.validation.engine.lar.LarCommonEngine
 import hmda.validation.rules.EditCheck
-import hmda.validation.rules.ts.syntactical.S025
+import hmda.validation.rules.ts.syntactical.{ S020, S025 }
 import hmda.validation.rules.lar.syntactical._
 
 trait LarSyntacticalEngine extends LarCommonEngine with ValidationApi {
 
   def checkSyntactical(lar: LoanApplicationRegister, ctx: ValidationContext): LarValidation = {
-    val checksToRun: List[EditCheck[LoanApplicationRegister]] = List(
+    val checksToRun: List[EditCheck[_ >: LoanApplicationRegister]] = List(
       S010,
       S020,
       S025.inContext(ctx),
       S205
     )
-    val checks = checksToRun.map(check(_, lar, lar.loan.id, Syntactical))
+    val checks = checksToRun.map(check[LoanApplicationRegister](_, lar, lar.loan.id, Syntactical))
```

Comment:
Yeah, the other LAR engines have only `EditCheck[LoanApplicationRegister]`s.

As for the difference between `LarSyntacticalEngine` and its `Ts` counterpart, it turns out to be subtle, and have to do with the declared types.  In the LAR version, that line of code is in a method that is declared to return a specific type. If you change the `checkSyntactical` return type to `ValidationNel[ValidationError, _ >: LoanApplicationRegister]`, then it compiles fine either way: with or without the explicit type in the `check` call.

Meanwhile in the TS version, those lines of code are in the context of a larger method that doesn't constrain its types (and specifically, the return type of the `validateAll` call) as much, and so things work out fine.

Contravariance in `EditCheck` is a better answer, but this question was educational and interesting.  I have a somewhat better sense for how the type inference is working here, now.